src/main/haskell/kore/src/Kore/ASTUtils/AlphaCompare.hs

```diff
@@ -0,0 +1,101 @@
+{-|
+Module      : Kore.ASTUtils.AlphaCompare
+Description : Compare
+Copyright   : (c) Runtime Verification, 2018
+License     : NCSA
+Maintainer  : phillip.harris@runtimeverification.com
+Stability   : experimental
+Portability : portable
+-}
+
+module Kore.ASTUtils.AlphaCompare
+    ( alphaCompare
+    ) where
+
+import Data.List
+
+import Kore.AST.Common
+import Kore.ASTUtils.SmartPatterns
+
+alphaCompare
+    :: (Eq (var level), Eq (var level))
+    => PureMLPattern level var
+    -> PureMLPattern level var
+    -> Bool
+alphaCompare e1 e2 = go [] [] e1 e2
+
+go
```

Comment:
Please move `go` into a `where` clause of `alphaCompare`:
1. The profiling report will be much easier to read.
1. The `Eq` constraint will not need to be passed over and over at runtime.